File: apps/runtime/container-images/python/healthcheck.py
#!/usr/bin/env python3
"""
Health check script for Python Cronium container
"""
import sys
import os

def main():
    """Perform basic health checks"""
    try:
        # Check if cronium module can be imported
        import cronium
        
        # Check environment variables are set
        required_env = ['CRONIUM_RUNTIME_API', 'CRONIUM_EXECUTION_TOKEN', 'CRONIUM_EXECUTION_ID']
        
        # In health check mode, these might not be set, so we just check the module loads
        if all(var in os.environ for var in required_env):
            # If env vars are set, verify the module initializes
            _ = cronium.Cronium()
        
        # Check Python version
        if sys.version_info < (3, 12):
            print(f"Python version too old: {sys.version}")
            sys.exit(1)
        
        print("Health check passed")
        sys.exit(0)
        
    except Exception as e:
        print(f"Health check failed: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()

File: apps/runtime/container-images/nodejs/healthcheck.js
#!/usr/bin/env node
/**
 * Health check script for Node.js Cronium container
 */

const { existsSync } = require("fs");
const { resolve } = require("path");

async function main() {
  try {
    // Check if cronium module can be loaded
    const croniumPath = resolve("/usr/local/lib/cronium.js");
    if (!existsSync(croniumPath)) {
      throw new Error("Cronium module not found");
    }

    // Try to require the module (without initializing since env vars might not be set)
    require(croniumPath);

    // Check Node.js version
    const nodeVersion = process.versions.node.split(".")[0];
    if (parseInt(nodeVersion) < 20) {
      throw new Error(`Node.js version too old: ${process.version}`);
    }

    // Check if axios is available
    try {
      require.resolve("axios");
    } catch (e) {
      throw new Error("axios module not found");
    }

    console.log("Health check passed");
    process.exit(0);
  } catch (error) {
    console.error(`Health check failed: ${error.message}`);
    process.exit(1);
  }
}

main();
